File: peanoware/java/peanoware/formula/Integer.java
/*
 * Integer.java
 *
 * Created on 14 ottobre 2004, 4.51
 */

package peanoware.formula;

/**
 *
 * @author  monica
 */
public class Integer {
    int i;
    
    /** Creates a new instance of Integer */
    public Integer() {
        i = 0;
    }
    public Integer(int i) {
        this.i = i;
    }   
    public int get() {
        return i;
    }
    public void decr() {
        i--;
    }
    public boolean isNull() {
        return i == 0;
    }
}

File: peanoware/javascript/Integer.js
/**
 * Integer.js
 *
 * Created on 14 ottobre 2004, 4.51
 * Translated to JavaScript
 *
 * @author monica
 */

class Integer {
  /**
   * Creates a new instance of Integer
   * @param {number} i - Initial value (defaults to 0)
   */
  constructor(i = 0) {
    this.i = i;
  }

  /**
   * Get the current value
   * @returns {number}
   */
  get() {
    return this.i;
  }

  /**
   * Decrement the value by 1
   */
  decr() {
    this.i--;
  }

  /**
   * Check if the value is zero
   * @returns {boolean}
   */
  isNull() {
    return this.i === 0;
  }
}

// Export for use in modules (optional)
// export default Integer;
